### PR TITLE
Add FirebaseRules resources

### PR DESCRIPTION
**Pull request by tylerg-dev**
Add Firebase Rules entities: Ruleset and Release.

This PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_firebaserules_ruleset`
```
```release-note:new-resource
`google_firebaserules_release`
```

### CODE REVIEW

**Comment by tylerg-dev** [comment]
The link provided for generated examples (https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) returns 404.  

Is there an alternate location I should be updating?

**Comment by ScottSuarez** [comment]
Looks like we're receiving a failure on the test with a failed import

```
=== RUN   TestAccFirebaserulesRelease_MinimalRelease
=== PAUSE TestAccFirebaserulesRelease_MinimalRelease
=== CONT  TestAccFirebaserulesRelease_MinimalRelease
    provider_test.go:286: Step 2/2 error running import: exit status 1
        There are some problems with the CLI configuration:
        
        Error: The specified plugin cache dir /opt/teamcity-agent/work/2e51954ade9939a/.provider-cache cannot be opened: stat /opt/teamcity-agent/work/2e51954ade9939a/.provider-cache: no such file or directory
        
        
        As a result of the above problems, Terraform may not behave as intended.
        
        
        
        Error: Cannot import non-existent remote object
        
        While attempting to import an existing object to
        google_firebaserules_release.primary, the provider detected that no object
        exists with the given id. Only pre-existing objects can be imported; check
        that the id is correct and that it is associated with the provider's
        configured region or endpoint, or use "terraform apply" to create a new remote
        object for this resource.
        
--- FAIL: TestAccFirebaserulesRelease_MinimalRelease (24.68s)
FAIL
```

**Comment by ScottSuarez** [comment]
given the other import passed the failure for the import doesn't make much sense

/gcbrun

going to try and rerun the suite

File: mmv1/third_party/terraform/utils/provider.go.erb [hunk 1]
```diff
@@ -468,6 +469,8 @@ end     # products.each do
 				"google_compute_firewall_policy_rule":          resourceComputeFirewallPolicyRule(),
 				"google_dataproc_workflow_template":            resourceDataprocWorkflowTemplate(),
 				"google_eventarc_trigger":                      resourceEventarcTrigger(),
+				"google_firebaserules_release":					resourceFirebaserulesRelease(),
+				"google_firebaserules_ruleset":					resourceFirebaserulesRuleset(),
```
[the remark this answers is not in the document]
Comment (tylerg-dev) [reply mmv1/third_party/terraform/utils/provider.go.erb]: Fixed.

File: mmv1/third_party/terraform/utils/provider.go.erb [hunk 2]
```diff
@@ -181,6 +181,7 @@ func Provider() *schema.Provider {
 			AssuredWorkloadsEndpointEntryKey:                AssuredWorkloadsEndpointEntry,
 			CloudResourceManagerEndpointEntryKey:            CloudResourceManagerEndpointEntry,
 			EventarcEndpointEntryKey:                        EventarcEndpointEntry,
+			FirebaserulesEndpointEntryKey:					 FirebaserulesEndpointEntry,
```
[the remark this answers is not in the document]
Comment (tylerg-dev) [reply mmv1/third_party/terraform/utils/provider.go.erb]: Fixed

**Comment by ScottSuarez** [comment]
hmm good question! So while the resource is able to be created it looks like import is failing.

It looks like when we try to parse the import we aren't expecting a `/` to be within the name parameter.
[modular-magician/terraform-provider-google-beta@`auto-pr-5695-old..auto-pr-5695`#diff-8380529ab3](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5695-old..auto-pr-5695#diff-8380529ab3c69694981b995e74cfa712108f39558a8671344f121c23e5b91772R281-R299)

If you look downstream at the import function it may become apparent on how to control these parameters to allow for this.
https://github.com/GoogleCloudPlatform/magic-modules/blob/master/tpgtools/templates/resource.go.tmpl#L548-L578

Let me know how things go.

**Comment by tylerg-dev** [comment]
AFAIK (based on local testing) https://github.com/GoogleCloudPlatform/magic-modules/pull/5775 addresses the import issue seen in the test.  Would re-running the tests demonstrate that, or do I need to take further steps first?

**Review by ScottSuarez — APPROVED** [review]
lgtm ! ~

**Pull request merged**